### ISSUE TITLE
Crash in [TEALTimestampDataSources stringOfTimestampAsISOLocalFrom:] with 5.4.3

### ISSUE THREAD

**Issue by JuhaO81** [Tealium/tealium-ios#83]
```
Thread 0 Crashed:
0   libobjc.A.dylib                      0x00000001ee370d68 objc_msgSend + 8
1   CoreFoundation                       0x00000001ef080b00 -[__NSFrozenDictionaryM objectForKey:] + 84
2   CoreFoundation                       0x00000001ef07d6f8 -[NSMutableDictionary addEntriesFromDictionary:] + 288
3   TealiumIOS                           0x0000000102703a94 -[Tealium captureTimeDataSourcesForType:title:] (Tealium.m:270)
4   TealiumIOS                           0x0000000102702e64 -[Tealium trackType:title:dataSources:completion:] (Tealium.m:164)
5   TealiumIOS                           0x0000000102702d38 -[Tealium trackEventWithTitle:dataSources:] (Tealium.m:140)
6   NBHouseholdFI                        0x0000000100e65cbc -[NBAnalyticsManager trackEvent:] (NBAnalyticsManager.m:490)
```

**Comment by JuhaO81** [comment]
same as: https://github.com/Tealium/tealium-ios/issues/73

**Comment by JuhaO81** [comment]
Our tester reported these steps:
-Changed device language into english and region into UK
-Tried to login to app

I cannot reproduce it with these steps but according to app store we keep having this crash every now and then. Not sure if above steps are correct but please check.

**Comment by jonathanswong** [comment]
Hi @JuhaO81 . We released 5.4.6 today that adds protection around volatile data sources being accessed from multiple threads. We'll keep this issue open in case this issue is still not fixed by the latest release. Thank you.

**Comment by JuhaO81** [comment]
Thx! I'm updating now. I'm not sure when is our next release window with this change but I let you know if issue still remains.
